Tetrahedral_remeshing/include/CGAL/Tetrahedral_remeshing/internal/tetrahedral_remeshing_helpers.h

```diff
@@ -762,12 +764,15 @@ Subdomain_relation compare_subdomains(const typename C3t3::Vertex_handle v0,
                                       const C3t3& c3t3)
 {
   typedef typename C3t3::Subdomain_index Subdomain_index;
+  typedef boost::container::flat_set<Subdomain_index,
+    std::less<Subdomain_index>,
+    boost::container::small_vector<Subdomain_index, 30> > Set_of_subdomains;
```

Comment:
We have an issue. That is only available [since Boost-1.66](https://www.boost.org/doc/libs/1_67_0/doc/html/container/release_notes.html#container.release_notes.release_notes_boost_1_66_00).

For the moment, we document that the minimal version of Boost is 1.62, but we actually only test from Boost-1.65 (on Gauguin, platform `x64_Cygwin-Windows10_MSVC2015-Debug-64bits`). Maybe we should bump the minimal version to 1.66.

@maxGimeno Is that easy to bump Boost on Gauguin?

Reply:
Does this show up in vtune.  If not isn't this over-optimization.   Or just best practice?

Reply:
Don't we use `small_vector` not already in other places?

Reply:
`small_vector` is from Boost-1.59. But the possibility to use `small_vector` as the storage of a `flat_set` is from Boost-1.66.

Are you are right: that is a micro-optimization. But we do not use VTune that much, and usually after using VTune, the usual optimizations are the use of `small_vector` to avoid memory allocations. So that is a micro-optimization that we have needed several times in the past. For me, that turns it into a best-practice: avoid allocating memory is a function that is called frequently.

Reply:
@lrineau No, it's not hard

Reply:
So I suggest we stop support 1.65, and say from CGAL 5.1 only Boost 1.66 or later is supported. That would be simpler.

Reply:
Travis fails on this PR because the version of Boost is too old.

Reply:
@maxGimeno You will have to update Boost on Travis Xenial.
You can use this PPA to install Boost 1.73:
https://launchpad.net/~mhier/+archive/ubuntu/libboost-latest?field.series_filter=xenial